### DISCUSSION TITLE
Running `queryRaw` with a parametric value for `nextVal` is not working

### DISCUSSION THREAD

**Discussion by steliosrammos** [prisma/prisma#17091] [Q&A]
I am trying to run the following raw query: 

```
const sequenceName = 'activity_id_seq';
const res = await db.$queryRawUnsafe('select nextval($1);', sequenceName) as { nextval: number }[];
```

Unfortunately it results in the following error: 

```
Raw query failed. Code: `22P03`. Message: `db error: ERROR: incorrect binary data format in bind parameter 1`
```

I have also tried putting the whole `nextVal('activity_id_seq')` as the parameter but no luck with that either.

**Prisma version:** 3.11.0
**Database:** PostrgreSQl 12.0

**Comment by nurul3101** [comment]
Hello @steliosrammos 👋 

Can you try updating Prisma to version 4 and check?
We made significant improvements for Raw queries in version 4, please let me know in case updating to version 4 also doesn't fix the issue for you.

Here's a [guide](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-4) on Upgrading to Prisma 4.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
